core/src/core-api/_internalLinkManager.js

```diff
@@ -54,6 +54,30 @@ export class linkManager extends LuigiCoreAPIBase {
     return remotePromise;
   }
 
+  navigateToIntent(semanticSlug, params = {}) {
+    let newPath = '#?intent=';
+
+    newPath += semanticSlug;
+
+    if (Object.keys(params)?.length) {
+      const paramList = Object.entries(params);
+
+      // append parameters to the path if any
+      if (paramList.length > 0) {
+        newPath += '?';
+
+        for (const [key, value] of paramList) {
+          newPath += key + '=' + value + '&';
+        }
+
+        // trim potential excessive ampersand & at the end
+        newPath = newPath.slice(0, -1);
+      }
+    }
+
+    this.navigate(newPath);
+  }
```

Comment:
I think we have unit tests for _internalLinkManager.js file https://github.com/SAP/luigi/blob/main/core/test/core-api/internal-link-manager.spec.js

Can you add unit tests for it there as well. We had no unit tests for this function previously as it was residing on LuigiClient. Would be nice to now test it here.